Firestore/core/src/firebase/firestore/remote/buffered_writer.cc

```diff
@@ -18,9 +18,6 @@
 
```

Comment:
@wilhuff Did yet more changes; I think this is in reasonable state for a high-level review now:
* since operations now delete themselves in `Complete`, I wanted to ensure that any created operation is always put on the GRPC queue (which guarantees that `Complete` will be called on them). To disallow other classes to create but not execute operations, I moved `MakeOperation`/`ExecuteOperation` from `GrpcStream` into `StreamOperation` (as a static function). The function creates a "raw" new operation, `Execute`s it immediately (which must put the operation on GRPC queue) and returns a non-owning pointer to the operation.
* the change from the previous bullet allowed changing `BufferedWriter` so that it can now create `StreamWrite` operations on its own. All (both) methods of `BufferedWriter` that may do a write now return a `StreamWrite` pointer if a new write is issued, or `nullptr` otherwise. Returning `StreamWrite` allows `GrpcStream` to store it in its `operations_` vector -- the idea is that `GrpcStream` is the only class that stores all currently executing operations.
* changed `WriteAndFinish` to be a blocking operation. Experiments show that in the normal case write operation gets off the GRPC queue in ~50-100 microseconds; in case there is no connection, the delay is slightly longer, but still less than a millisecond. Still, in case of unforeseen circumstances, I added a timeout to the wait.